fixbackend/cloud_accounts/service_impl.py

```diff
@@ -361,6 +411,36 @@ def compute_failed_scan_count(acc: CloudAccount) -> int:
                 )
                 await send_pub_sub_message(degraded_event)
 
+            case ProductTierChanged.kind:
+                ptc_evt = ProductTierChanged.from_json(message)
+                new_account_limit = ProductTierSettings[ptc_evt.product_tier].account_limit or math.inf
+                old_account_limit = ProductTierSettings[ptc_evt.previous_tier].account_limit or math.inf
+                if new_account_limit < old_account_limit:
+                    # we should not have infinity here
+                    new_account_limit = round(new_account_limit)
+                    # tier changed, time to delete accounts
+                    all_accounts = await self.list_accounts(ptc_evt.workspace_id)
+                    # keep the last new_account_limit accounts
+                    to_delete = all_accounts[:-new_account_limit]
+                    # delete them all in parallel
+                    async with asyncio.TaskGroup() as tg:
+                        for cloud_account in to_delete:
+                            tg.create_task(
+                                self.delete_cloud_account(ptc_evt.user_id, cloud_account.id, ptc_evt.workspace_id)
+                            )
+
+            case AwsMarketplaceSubscriptionCancelled.kind:
+                evt = AwsMarketplaceSubscriptionCancelled.from_json(message)
+                workspaces = await self.workspace_repository.list_workspaces_by_subscription_id(evt.subscription_id)
+                async with asyncio.TaskGroup() as tg:
+                    for ws in workspaces:
+                        account_limit = Free.account_limit or 1
+                        all_accounts = await self.list_accounts(ws.id)
+                        # keep the last account_limit accounts
+                        to_disable = all_accounts[:-account_limit]
+                        for cloud_account in to_disable:
+                            tg.create_task(self.update_cloud_account_enabled(ws.id, cloud_account.id, False))
```

Comment:
When the subscription is cancelled, all accounts get disabled. Fine.
What prevents the user from enabling it again?